### ISSUE TITLE
WebpackError: Unsafe builtin usage http.Agent

### ISSUE THREAD

**Issue by beamercola** [algolia/algoliasearch-client-javascript#1259]
I'm using this library in Gatsby 3 an I'm getting the following error:

```
warn WebpackError: Unsafe builtin usage http.Agent
    at my-website/node_modules/@algolia/requester-node-http/dist/requester-node-http.esm.js:9:1
    at my-website/webpack/bootstrap:18:1
    at my-website/node_modules/algoliasearch/dist/algoliasearch.cjs.js:10:25
    at my-website/webpack/bootstrap:18:1
    at my-website/node_modules/algoliasearch/index.js:2:23
    at my-website/webpack/bootstrap:18:1
    at my-website/webpack/bootstrap:18:1
    at my-website/.cache/_this_is_virtual_fs_path_/$virtual/sync-requires.js:11:52
    at my-website/webpack/bootstrap:18:1
    at my-website/.cache/static-entry.js:11:22
    at my-website/node_modules/@algolia/client-search/dist/client-search.esm.js:512:10
    at my-website/node_modules/@algolia/client-search/dist/client-search.esm.js:540:1
    at my-website/node_modules/@algolia/client-search/dist/client-search.esm.js:376:1

   7 | /* eslint functional/prefer-readonly-type: 0 */
   8 | const agentOptions = { keepAlive: true };
>  9 | const defaultHttpAgent = new Agent(agentOptions);
     | ^
  10 | const defaultHttpsAgent = new Agent$1(agentOptions);
  11 | function createNodeHttpRequester({ agent: userGlobalAgent, httpAgent: userHttpAgent, httpsAgent: userHttpsAgent, requesterOptions = {}, } = {}) {
  12 |     const httpAgent = userHttpAgent || userGlobalAgent || defaultHttpAgent;
```

**Comment by mikewojo** [comment]
No idea how to solve this, but believe it's related to:

> Gatsby v3 introduces incremental builds for HTML generation. For this feature to work correctly, Gatsby needs to track all inputs used to generate HTML file. Arbitrary code execution in gatsby-ssr.js files allow usage of fs module, which is marked as unsafe and results in disabling of this feature.

https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#using-fs-in-ssr

Because the full warning text is:
```
warn Unsafe builtin method was used, future builds will need to rebuild all pages
warn WebpackError: Unsafe builtin usage http.Agent
```

**Comment by beamercola** [comment]
@mikewojo mama mia it works.  large shopify + prismic site building in 45 seconds.  let's go gatsby 3!

**Comment by graysonhicks** [comment]
For future readers, [this link](https://www.gatsbyjs.com/docs/debugging-html-builds/#fixing-third-party-modules) describes the other way to fix this. (loadable-components being the other)

```javascript
exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
  if (stage === "build-html" || stage === "develop-html") {
    actions.setWebpackConfig({
      module: {
        rules: [
          {
            test: /algoliasearch/,
            use: loaders.null(),
          },
        ],
      },
    })
  }
}
```

**Comment by FullStackTiger** [comment]
solved it on algoliasearch version "^4.10.2" with gatsby version 3.0
